rpc/gnmi/gnmi-commit-confirmed.md

```diff
@@ -0,0 +1,68 @@
+# gNMI Commit Confirmed Extension
+
+**Contributors:**
+
+**Date:** October 6, 2023
+
+**Version:** 0.1.0
+
+## 1. Purpose
+
+In certain deployments, client and server is separated by a complex network, hence we cannot assume
+
+- The pushed configuration will not break connectivity to the network device.
+- The network device has out-of-band access.
+
+This extension provides a way for a controller/operator to apply updates in a `SetRequest` that will require a subsequent confirmation in a time window specified by the `timeout` value.
+
+If no confirmation is received within the time window, the network device will revert the changes restoring the previous state. This workflow is particularly useful to ensure that erroneous configuration will not break connectivity to the network device.
+
+## 2. Definition
+
+A [`CommitConfirmed`](TBD) message is added to the list of well-known gNMI Extensions and is defined in [gnmi_ext.proto](TBD).
+
+The message contains the following fields:
+
+- `action` - The action to be taken by the network device upon receiving the `SetRequest` with the `CommitConfirmed` extension.
+- `timeout` - time in seconds/nanoseconds to wait for a confirmation before reverting the changes.
+- `commit_id` - commit identifier returned by the network device that is associated with the commit operation.
+
+### 3. Set Request handling
+
+#### 3.1 Initiating Commit Confirmed operation
+
+To initiate a Commit Confirmed operation, the client MUST include the `CommitConfirmed` extension in the `SetRequest` message. The `CommitConfirmed` message MUST include the `COMMIT_CONFIRMED_ACTION_START` action to indicate that the network device MUST apply the configuration changes and start the commit rollback timeout waiting for a confirmation action.
```

Comment:
We need to specify the locking here - i.e. the commit to commit confirm (or abort or timeout) is a window in which the server is locked from the perspective of other sessions.  I would prefer this to be explicit in the ACTION naming such as START_W_LOCK/CONFIRM_W_UNLOCK/REJECT_W_UNLOCK which may push towards:
ACTION_LOCK (instead of start), ACTION_UNLOCK_REJECT, ACTION_UNLOCK_CONFIRM.

Failing this being explicit in the actions it should be very clear in this description and in the proto etc.